omnigibson/objects/controllable_object.py

```diff
@@ -428,21 +429,36 @@ def deploy_control(self, control, control_type, indices=None, normalized=False):
                 ctrl = control[cur_ctrl_idx]
 
             # Deploy control based on type
-            ctrl_type, norm = control_type[cur_ctrl_idx], normalized[cur_ctrl_idx]       # In multi-DOF joint case all values were already checked to be the same
+            ctrl_type = control_type[cur_ctrl_idx]       # In multi-DOF joint case all values were already checked to be the same
             if ctrl_type == ControlType.EFFORT:
-                joint.set_effort(ctrl, normalized=norm)
+                eff_vec.append(ctrl)
+                eff_idxs.append(cur_ctrl_idx)
+                using_eff = True
             elif ctrl_type == ControlType.VELOCITY:
-                joint.set_vel(ctrl, normalized=norm, drive=True)
+                vel_vec.append(ctrl)
+                vel_idxs.append(cur_ctrl_idx)
+                using_vel = True
             elif ctrl_type == ControlType.POSITION:
-                joint.set_pos(ctrl, normalized=norm, drive=True)
+                pos_vec.append(ctrl)
+                pos_idxs.append(cur_ctrl_idx)
+                using_pos = True
             elif ctrl_type == ControlType.NONE:
                 # Set zero efforts
-                joint.set_effort(0, normalized=False)
+                eff_vec.append(0)
+                eff_idxs.append(cur_ctrl_idx)
+                using_eff = True
             else:
                 raise ValueError("Invalid control type specified: {}".format(ctrl_type))
-
             # Finally, increment the current index based on how many DOFs were just controlled
             cur_indices_idx += joint_dof
+            
+        # set the targets for joints
+        if using_pos:
+            self.set_joint_positions(positions=np.array(pos_vec), indices=np.array(pos_idxs), drive=True, normalized=normalized)
+        if using_vel:
+            self.set_joint_velocities(velocities=np.array(vel_vec), indices=np.array(vel_idxs), drive=True, normalized=normalized)
+        if using_eff:
+            self.set_joint_efforts(efforts=np.array(eff_vec), indices=np.array(eff_idxs), normalized=normalized)
```

Comment:
From what I can tell, this doesn't actually do much other than possibly a minor speedup?

Reply:
The change is made to get the osc working. Basically it seems like if we set the joint effort individually when we set the latter ones the previous values will get overwritten

Reply:
That doesn't sound right. Please file an issue before you merge.

Reply:
issue filed